src/handler/mod.rs

```diff
@@ -0,0 +1,111 @@
+/// This module contains most of the low-level commands
+/// and grub variable modifications
+use anyhow::{bail, Result};
+
+use std::process::Command;
+use std::str;
+
+/// reboots the system if boot_counter is greater than 0 or can be forced too
+pub fn handle_reboot(force: bool) -> Result<()> {
+    if !force {
+        if let Some(t) = get_boot_counter()? {
+            if t <= 0 {
+                bail!("countdown ended, check greenboot-rollback status")
+            };
+        }
+    }
+    log::info!("restarting the system");
+    Command::new("systemctl").arg("reboot").status()?;
+    Ok(())
+}
+
+/// rollback to previous ostree deployment if boot counter is less than 0
+pub fn handle_rollback() -> Result<()> {
+    if let Some(boot_counter) = get_boot_counter()? {
+        if boot_counter <= 0 {
+            log::info!("Greenboot will now attempt to rollback");
+            Command::new("rpm-ostree").arg("rollback").status()?;
+            return Ok(());
+        }
+    }
+    bail!("Rollback not initiated");
+}
```

Comment:
this can be better rewritten as:

```
pub fn handle_rollback() -> Result<()> {
    boot_counter = get_boot_counter()?;
    if boot_counter <= 0 {
        log::info!("Greenboot will now attempt to rollback");
        Command::new("rpm-ostree").arg("rollback").status()?;
        return Ok(());
    }
    bail!("Rollback not initiated");
}
```

Reply:
fixed in : ee8bd08b11f22f930ca7e0e7a43733cbd28dd8b2